app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def configure_permitted_parameters
+    # For additional fields in app/views/devise/registrations/new.html.erb
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:first_name, :last_name, :mobile_number, :address])
+
+    # For additional in app/views/devise/registrations/edit.html.erb
+    devise_parameter_sanitizer.permit(:account_update, keys: [:fist_name, :mobile_number])
```

Comment:
you wouldn't have here some parameters that you have already included above (ie first name and mobile numbers are already included in the line above. Not sure we need to give anything after 'keys' here, not even sure if we need that line but lets keep in in case